### ISSUE TITLE
Feature Request: Format file in-place

### ISSUE THREAD

**Issue by swateek** [sql-formatter-org/sql-formatter#406]
**Description**
It'll be a good addition to have support for adding this to  [pre-commit](https://pre-commit.com/) 
Could we add support for that?

**Why do you want this feature?**
Formatting SQL is a good idea before committing code to repo

**Comment by nene** [comment]
Well, I'm completely unfamiliar with this pre-commit tool, but I have nothing against it using sql-formatter in however it pleases.

So, what's the actual problem? Why can't this library currently be used with that pre-commit tool?

**Comment by swateek** [comment]
I am sorry, I assumed it to be of common knowledge.

`pre-commit` is a super popular tool in the Python (and DataScience)
community to format SQL scripts.

I came across this since this is currently the only library which formats
SingleStore files.


What it takes to support?

Take a look at sql-fluff, they have two pre-commit configuration files in
their project home which helps in running the library as a pre-commit hook.


I have forked this repo, will take a stab at this.

**Comment by inferrinizzard** [comment]
Since this is a typescript library on the node ecosystem, I assumed you were referring to the npm [pre-commit ](https://www.npmjs.com/package/pre-commit), which would easily support this.

As for the python pre-commit, this use case is unlikely to be supported since we're mixing ecosystems here but you can also just call the library via cli through the `node` (add a `package.json`) or `bash` (use `npx`) plugins

**Comment by slyapustin** [comment]
> So, what's the actual problem? Why can't this library currently be used with that pre-commit tool?

@nene I'm late to the party, but I spent some time today trying to integrate sql-formatter with the pre-commit.com.
Here is a new repository I created: https://github.com/slyapustin/pre-commit-sql-formatter

The only things which might be really helpful and might simplify things a lot is a rewrite option.

Basically pre-commit invoke hooks like this:
> hook-command <filename>

So in case of the sql-foramatter I would like to have arguments, which if provided can format file and save it back to the same location. Not that pre-commit only pass filename once in the end of the hook-command.

Here is that I would like to run (note that file name must be passed once in the end):
> sql-format <some options> my-app/my-cool-query.sql

And I expect reformatted `my-app/my-cool-query.sql` being reformatted and saved in to the same location.
Hope that helps.

**Comment by nene** [comment]
This somewhat relates to another feature request (#552), but is a lot simpler.

Basically all that's needed is to have an alternative syntax for what can be currently done with specifying the same file as both input and output:

```
sql-formatter --output example.sql example.sql
```

For example:

```
sql-formatter --write example.sql
# or
sql-formatter --replace example.sql
# or
sql-formatter --fix example.sql
```

This should be pretty simple to accomplish. Not entirely sure which name would be best, but I'm currently leaning towards `--fix`.

**Comment by slyapustin** [comment]
Hey @nene!

Yes, that's exactly what we need!
I like using `--fix` for that.

Thanks!

**Comment by nene** [comment]
The `--fix` flag is now available in 12.1.0 release 🚀

**Comment by slyapustin** [comment]
@nene Thank you so much! It works great. 
I went ahead and created PR #556 to be able directly integrates with `sql-formatter` from the `pre-commit`.
Please take a look.
